src/synapse/workload.py

```diff
@@ -138,6 +138,105 @@ def check_mjolnir_ready() -> ops.pebble.CheckDict:
     return check.to_dict()
 
 
+def _get_configuration_field(container: ops.Container, fieldname: str) -> typing.Optional[str]:
+    """Get configuration field.
+
+    Args:
+        container: Container of the charm.
+        fieldname: field to get.
+
+    Raises:
+        PathError: if somethings goes wrong while reading the configuration file.
+
+    Returns:
+        configuration field value.
+    """
+    try:
+        configuration_content = str(container.pull(SYNAPSE_CONFIG_PATH, encoding="utf-8").read())
+        value = yaml.safe_load(configuration_content)[fieldname]
+        return value
+    except PathError as path_error:
+        if path_error.kind == "not-found":
+            logger.debug(
+                "configuration file %s not found, will be created by config-changed",
+                SYNAPSE_CONFIG_PATH,
+            )
+            return None
+        logger.exception(
```

Comment:
Won't this log message also include the one in line 159?

Reply:
Since there is a return None in case of "not-found", I believe it won't be included. 

And my bad: I sorted this file while working on this PR but most of these functions are unrelated to the SMTP configuration and could be reviewed in the Synapse Charm Review PR.